certbot-apache/certbot_apache/configurator.py

```diff
@@ -1943,79 +1890,7 @@ def cleanup(self, achalls):
         if not self._chall_out:
             self.revert_challenge_config()
             self.restart()
-            self.parser.init_modules()
-
-
-def _get_mod_deps(mod_name):
-    """Get known module dependencies.
-
-    .. note:: This does not need to be accurate in order for the client to
-        run.  This simply keeps things clean if the user decides to revert
-        changes.
-    .. warning:: If all deps are not included, it may cause incorrect parsing
-        behavior, due to enable_mod's shortcut for updating the parser's
-        currently defined modules (`.ApacheConfigurator._add_parser_mod`)
-        This would only present a major problem in extremely atypical
-        configs that use ifmod for the missing deps.
-
-    """
-    deps = {
-        "ssl": ["setenvif", "mime"]
-    }
-    return deps.get(mod_name, [])
-
-
-def get_file_path(vhost_path):
-    """Get file path from augeas_vhost_path.
-
-    Takes in Augeas path and returns the file name
-
-    :param str vhost_path: Augeas virtual host path
-
-    :returns: filename of vhost
-    :rtype: str
-
-    """
-    if not vhost_path or not vhost_path.startswith("/files/"):
-        return None
-
-    return _split_aug_path(vhost_path)[0]
-
-
-def get_internal_aug_path(vhost_path):
-    """Get the Augeas path for a vhost with the file path removed.
-
-    :param str vhost_path: Augeas virtual host path
-
-    :returns: Augeas path to vhost relative to the containing file
-    :rtype: str
-
-    """
-    return _split_aug_path(vhost_path)[1]
-
-
-def _split_aug_path(vhost_path):
-    """Splits an Augeas path into a file path and an internal path.
-
-    After removing "/files", this function splits vhost_path into the
-    file path and the remaining Augeas path.
-
-    :param str vhost_path: Augeas virtual host path
-
-    :returns: file path and internal Augeas path
-    :rtype: `tuple` of `str`
-
-    """
-    # Strip off /files
-    file_path = vhost_path[6:]
-    internal_path = []
-
-    # Remove components from the end of file_path until it becomes valid
-    while not os.path.exists(file_path):
-        file_path, _, internal_path_part = file_path.rpartition("/")
-        internal_path.append(internal_path_part)
-
-    return file_path, "/".join(reversed(internal_path))
+            self.parser.reset_modules()
```

Comment:
It is actually done here: https://github.com/certbot/certbot/blob/373914fe719c50ca8474ca867e059b57eaeb6b80/certbot-apache/certbot_apache/configurator.py#L1891

But we just want to reset the parser state, ie: reading the active modules again from the configuration (and optionally from binary config dump).

Reply:
Sorry my comment wasn't very clear. I was referring to the `IInstaller` methods that revert changes made to the Apache config using the reverter class. Since we're rolling back the config to a previous state, the loaded modules may have changed. This is `rollback_checkpoints` and `recovery_routine` methods.

Reply:
Ah, yeah. Ultimately we should,  that should most likely be done at the same time when/if we restructure the ApacheConfigurator <-> AugeasConfigurator <-> ApacheParser relations.

This is currently the only runtime rollback we're doing, so it's not very urgent.

Reply:
I agree it's not necessary with the way Certbot currently uses its plugins, but this could cause bugs if this changes in the future. We should make an issue so we don't forget about this.

Reply:
I created issue #5227 to track this.